kork-core/src/main/java/com/netflix/spinnaker/kork/spring/PluginLoader.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2019 Armory, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.kork.spring;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class PluginLoader {
+
+  static final String DEFAULT_PLUGIN_CONFIG_PATH = "/opt/spinnaker/config/plugins.yml";
+
+  private String filename;
+  private List<URL> urls;
+
+  PluginLoader(String filename) {
+    this.filename = filename;
+  }
+
+  PluginLoader() {
+    this(
+        Optional.ofNullable(System.getenv("PLUGIN_CONFIG_LOCATION"))
+            .orElse(PluginLoader.DEFAULT_PLUGIN_CONFIG_PATH));
+  }
+
+  public void loadPlugins(Class source) throws PluginLoaderException {
```

Comment:
Looks like `PluginLoaderException` is a RuntimeException. You don't need to declare (throws) it.